server/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -671,4 +671,19 @@ public static boolean isArrayOfNulls(DataType<?> type) {
     public static boolean isNestedArray(DataType<?> type) {
         return type instanceof ArrayType<?> at && at.innerType() instanceof ArrayType<?>;
     }
+
+    public static DataType<?> deparametrize(DataType<?> type) {
+        if (type.id() == ArrayType.ID) {
+            return new ArrayType<>(deparametrize(((ArrayType<?>) type).innerType()));
+        } else if (type.id() == ObjectType.ID) {
+            var builder = ObjectType.builder();
+            for (var innerType : ((ObjectType) type).innerTypes().entrySet()) {
+                builder.setInnerType(innerType.getKey(), deparametrize(innerType.getValue()));
+            }
+            return builder.build();
+        } else {
+            type = DataTypes.ofName(type.getTypeSignature().getBaseTypeName());
```

Comment:
It fixes `'abc'::char(3) = 'abc'::char(5)` as well but breaks bitstrings because bitstrings length is 1 by default.